src/slg/scene/sceneobjectdefs.cpp

```diff
@@ -81,12 +85,26 @@ void SceneObjectDefinitions::UpdateMaterialReferences(const Material *oldMat, co
 		static_cast<SceneObject *>(o)->UpdateMaterialReferences(oldMat, newMat);
 }
 
-void SceneObjectDefinitions::UpdateMeshReferences(const ExtMesh *oldMesh, ExtMesh *newMesh,
-		boost::unordered_set<SceneObject *> &modifiedObjsList) {
-	for (auto o : objs.GetObjs()) {
-		SceneObject *so = static_cast<SceneObject *>(o);
-
-		if (so->UpdateMeshReference(oldMesh, newMesh))
-			modifiedObjsList.insert(so);
-	}
+void SceneObjectDefinitions::UpdateMeshReferences(const ExtMesh* oldMesh, ExtMesh* newMesh,
+	boost::unordered_set<SceneObject*>& modifiedObjsList) {
+
+	auto p = meshToSceneObjects.equal_range(oldMesh->GetName());
+	for (auto it = p.first; it != p.second; ++it)
+	{
+		std::string soName = it->second;
+		if (objs.IsObjDefined(soName))
+		{
+			SceneObject* so = static_cast<SceneObject*>(objs.GetObj(soName));
+			if (so->UpdateMeshReference(oldMesh, newMesh))
+			{
+				modifiedObjsList.insert(so);
+				
+				// change index
+				meshToSceneObjects.insert(make_pair(newMesh->GetName(), so->GetName()));
+			}
+		}
+		meshToSceneObjects.erase(it);
```

Comment:
Should the object be erased in any case? I think this should be the case only if the mesh is updated, i.e. in the if (so->UpdateMeshReference(oldMesh, newMesh)) codeblock after line 103.

Reply:
this method have problems i see.. sometimes crash the render in live session. in theory method want to replace all oldmesh references in scene objects with the new one so it is correct to erase all the old entry in the index structure .. ad add this method for debug and it seems to work good but some time still crash. 
I resolve the situation for now commenting the method UpdateMeshReferences , in  live session in unuseful but with one  million of instances it slow down the performance.

std::string SceneObjectDefinitions::ToString() const
{
	stringstream ss;

	ss << "[\n";

	// begin() returns iterator to first element of map
	auto it = meshToSceneObjects.begin();

	for (; it != meshToSceneObjects.end(); it++) {
		ss << "<" << it->first << ", " << it->second
			<< ">  ";
		ss << endl;
	}
	return ss.str();
}

Reply:
Now, it erase all entries in the meshToSceneObjects list and only add new entries in the conditional code block of 
`if (so->UpdateMeshReference(oldMesh, newMesh))`. I think that this is wrong as there are missing entries for the up to date references and this leads to the crash.
When you want to erase the entry in the meshToSceneObjects only if an old mesh reference is updated, then you simply have to put it in the conditional codeblock, i.e. under line 103.